service/scheduler.go

```diff
@@ -130,18 +147,28 @@ func (s *Scheduler) createActivateWorkflow(
 		return s.deleteWorkflows(client, createdIDs, statuses)
 	}
 	for _, createdID := range createdIDs {
-		err = client.Activate(createdID)
-		if err != nil {
-			fmt.Printf("[error] error activating workflow: %s\n", err)
-			notifyOwner(wf, err)
-			return s.cancelWorkflows(client, statuses)
-		}
-		statuses[createdID] = Activated.ToString()
+		statuses[createdID] = Created.ToString()
+	}
+	return statuses
+}
+
+func (s *Scheduler) activateWorkflow(
+	client *orchard.OrchardRestClient,
+	swf table.ScheduledWorkflow,
+	wf table.Workflow,
+) map[string]string {
+	statuses := make(map[string]string)
+	statuses[swf.OrchardID] = swf.Status
+	if err := client.Activate(swf.OrchardID); err != nil {
+		fmt.Printf("[error] error activating workflow: %s\n", err)
+		notifyOwner(wf, err)
+		return s.cancelWorkflows(client, statuses)
```

Comment:
with delayed activation, we probably don't need atomicity on activation anymore

Reply:
actually, I was confused with the `statuses` here, maybe now it can just return one single status? so that it's not confusing

Reply:
so represent all sub-workflows with one `status` for the parent workflow?

Reply:
returning just one single status now.

Reply:
@damianxu88 in the new decoupled mechanism for create / activate.  Activate handles one workflow at a time, each of the potentially more than 1 created workflows.  This return status for activate corresponds to just 1 workflow.

Reply:
@schowsf ah gotcha